File: dotnet/src/Experimental/Process.Abstractions/KernelProcessMapState.cs
// Copyright (c) Microsoft. All rights reserved.

using System.Runtime.Serialization;

namespace Microsoft.SemanticKernel;

/// <summary>
/// Represents the state of a <see cref="KernelProcessMap"/>.
/// </summary>
[DataContract]
public sealed record KernelProcessMapState : KernelProcessStepState
{
    /// <summary>
    /// Initializes a new instance of the <see cref="KernelProcessMapState"/> class.
    /// </summary>
    /// <param name="name">The name of the associated <see cref="KernelProcessMap"/></param>
    /// <param name="version">version id of the process step state</param>
    /// <param name="id">The Id of the associated <see cref="KernelProcessMap"/></param>
    public KernelProcessMapState(string name, string version, string id)
        : base(name, version, id)
    {
        Verify.NotNullOrWhiteSpace(id, nameof(id));
    }
}

File: python/semantic_kernel/processes/kernel_process/kernel_process_map_state.py
from semantic_kernel.processes.kernel_process.kernel_process_step_state import KernelProcessStepState


class KernelProcessMapState(KernelProcessStepState):
    """The state of a map step in a kernel process."""

    def __init__(self, name: str, version: str, id: str) -> None:
        """Initializes a new instance of the KernelProcessMapState class."""
        super().__init__(name, version, id)
